### PR TITLE
Improve RBAC documentation

### LINKED ISSUE #240: Improve RBAC documentation
## Overview

Yesterday during our RBAC-related conversation, @donnalux and @brampling spotted missing stuff in our docs. It would be great to address them soon so that it is clear to the user.

## Acceptance Criteria

Update RBAC doc (https://docs.botkube.io/configuration/rbac) not only for `next`, but also for `1.1`,`1.2`, `1.3`:

- describe explicitly all possible mappings:
   - user: static or channel name
   - group: static or channel name
- describe default permissions (`botkube-plugins-default` - https://github.com/kubeshop/botkube/blob/main/helm/botkube/values.yaml#L53C40-L53C40)

## Reason

Currently it's hard to get a full picture of the RBAC mappings as we don't mention e.g. group static mapping, or what are the defaults.

### PR DESCRIPTION
**Pull request by pkosiec**
## Description

Changes proposed in this pull request:

- Describe explicitly all possible mappings:
- Describe defaults
- Improve wording and structure of the document

## Preview

- https://rbac-config.botkube-docs-dcb.pages.dev/next/configuration/rbac
- https://rbac-config.botkube-docs-dcb.pages.dev/configuration/rbac

## Related issue(s)

Resolves #240

### CODE REVIEW

File: docs/configuration/rbac.md [hunk 1]
```diff
@@ -4,252 +4,449 @@ title: RBAC
 sidebar_position: 4
 ---
 
-Botkube allows users to restrict plugins by defining [RBAC](https://en.wikipedia.org/wiki/Role-based_access_control) rules as part of plugin configuration.
+Botkube allows plugins to access Kubernetes API by defining [RBAC](https://en.wikipedia.org/wiki/Role-based_access_control) rules as part of plugin `context.rbac` configuration. Kubeconfig generation needs to be requested by defining `context.rbac` property.
 
-Based on this configuration Botkube generates a temporary [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) file with user/group impersonation.
-This kubeconfig is available to plugins in the `Execute` and `Stream` contexts. If you're a plugin developer and want to learn more, refer to [Plugin Development](../plugin/using-kubeconfig.md) docs.
+Based on this configuration Botkube generates a temporary [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) file with user and/or group impersonation.
```
Comment (mszostok) [inline docs/configuration/rbac.md:9]:
```suggestion
Based on this configuration Botkube generates a temporary [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) with user and/or group impersonation.
```

File: docs/configuration/rbac.md [hunk 2]
```diff
@@ -4,252 +4,449 @@ title: RBAC
...
-
-Botkube uses its own cluster credentials to generate a temporary kubeconfig, and the temporary kubeconfig
-only [impersonates](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation) the requested user/group.
+Botkube uses its own cluster credentials to generate a temporary kubeconfig, and the kubeconfig uses [user/group impersonation](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation).
 
 For source plugins, the kubeconfig is generated once - during plugin startup.
-For executor plugins, the kubeconfig is generated every time a command is sent to the plugin, which allows
-for greater flexibility, such as including the name of the channel the command was sent from in the
+For executor plugins, the kubeconfig is generated every time a command is sent to the plugin, which allows greater flexibility, such as including the name of the channel the command was sent from in the
 kubeconfig generation.
```
Comment (mszostok) [inline docs/configuration/rbac.md:18]:
```suggestion
For executor plugins, the kubeconfig is generated every time a command is sent to the plugin, which allows greater flexibility, such as including the name of the channel the command was sent from.
```

File: docs/configuration/rbac.md [hunk 3]
```diff
@@ -4,252 +4,449 @@ title: RBAC
...
 
 ![diagram](assets/botkube-read-only.svg "diagram")
 
-### Configuration
+## Configuration
+
+Each executor and source plugin can provide a `context` section with `rbac` config. This config is used to generate a dedicated kubeconfig.
```
Comment (mszostok) [inline docs/configuration/rbac.md:24]:
```suggestion
For each executor and source plugin, you can define a `context.rbac` configuration.
```
it doesn't sound right for me when we tell that "plugin can" 🤔

File: docs/configuration/rbac.md [hunk 4]
```diff
@@ -4,252 +4,449 @@ title: RBAC
...
+
+```yaml
+rbac:
+  # ...
+  create: true
+  rules: # rules for the default static group
+    - apiGroups: ["*"]
+      resources: ["*"]
+      verbs: ["get", "watch", "list"]
+  staticGroupName: &static-group-name "botkube-plugins-default" # group name used in the `values.yaml` file for default plugins
```
Comment (mszostok) [inline docs/configuration/rbac.md:87]:
```suggestion
  staticGroupName: "botkube-plugins-default" # group name used in the `values.yaml` file for default plugins
```

File: docs/configuration/rbac.md [hunk 5]
```diff
@@ -4,252 +4,449 @@ title: RBAC
...
-#### Example: kubectl executor plugin with channel name mapping
+This paragraph contains examples of RBAC configuration for different use cases.
+
+:::tip
+You can use `extraObjects` override during Botkube installation. See the [`values.yaml`](https://github.com/kubeshop/botkube/blob/v1.3.0/helm/botkube/values.yaml#L1059) for more details.
+:::
 
-In this example **kubectl** executor plugin is configured with channel name mapping and bound to two channels,
-`ch-1` and `ch-2`.
+### Use default RBAC configuration
```
Comment (mszostok) [inline docs/configuration/rbac.md:146]: TBH I do not understand this section. Also, the padding is wrong, and the comma is missing.
Screenshot 2023-08-22 at 12 34 50

I would remove it, or we need to rephrase it. BTW above, you already have a section with:

``````markdown
executors:
  "kubectl":
    botkube/kubectl@v1:
      enabled: true
      # ...
      context:
        rbac: {} # enable kubeconfig generation
```

Here we define a plugin with _Static_ mapping to **User.rbac.authorization.k8s.io** `kubectl-read-only`.
However, such configuration will generate a kubeconfig with empty impersonation config, which effectively means an anonymous access to the Kubernetes API.
``````

File: docs/configuration/rbac.md [hunk 6]
```diff
@@ -4,252 +4,449 @@ title: RBAC
...
+     kind: ClusterRoleBinding
+     metadata:
+       name: reader
+     roleRef:
+       apiGroup: rbac.authorization.k8s.io
+       kind: ClusterRole
+       name: reader
+     subjects:
+     - kind: User
+       name: kubernetes-read-only
```
Comment (mszostok) [inline docs/configuration/rbac.md:259]:
```suggestion
       name: kubernetes-read-only # <-- this is the username used in Botkube config
```

to match the previous section

**Review by mszostok — APPROVED** [review]
Please address comments before merging. Thanks!

File: docs/configuration/rbac.md (hunk 4 above)
Comment (pkosiec) [reply docs/configuration/rbac.md:87]: I thought about it, and initially removed the anchor, but then I restored it to be closer to the `values.yaml` content 🤔 However I don't have strong opinion here, so I'll remove it 👍

File: docs/configuration/rbac.md (hunk 5 above)
Comment (pkosiec) [reply docs/configuration/rbac.md:146]: Right, I forgot to remove it as I introduced it above 🙈

**Review by mszostok — APPROVED** [review]

**Pull request merged**